src/webgpu/web_platform/reftests/canvas_display_after_device_lost.html.ts

```diff
@@ -0,0 +1,73 @@
+import { timeout } from '../../../common/util/timeout.js';
+import { assert } from '../../../common/util/util.js';
+import { takeScreenshotDelayed } from '../../../common/util/wpt_reftest_wait.js';
+
+void (async () => {
+  assert(
+    typeof navigator !== 'undefined' && navigator.gpu !== undefined,
+    'No WebGPU implementation found'
+  );
+
+  const adapter = await navigator.gpu.requestAdapter();
+  assert(adapter !== null);
+  const device = await adapter.requestDevice();
+  assert(device !== null);
+  const presentationFormat = navigator.gpu.getPreferredCanvasFormat();
+  let deviceLost = false;
+
+  function draw(
+    canvasId: string,
+    alphaMode: GPUCanvasAlphaMode,
+    {
+      reconfigureAfterLost,
+      drawAfterLost,
+    }: { reconfigureAfterLost: boolean; drawAfterLost: boolean }
+  ) {
+    const canvas = document.getElementById(canvasId) as HTMLCanvasElement;
+    const ctx = canvas.getContext('webgpu') as unknown as GPUCanvasContext;
+    if (!deviceLost || reconfigureAfterLost) {
+      ctx.configure({ device, format: presentationFormat, alphaMode });
+    }
+
+    if (!deviceLost || drawAfterLost) {
+      const colorAttachment = ctx.getCurrentTexture();
+      const colorAttachmentView = colorAttachment.createView();
+
+      const encoder = device.createCommandEncoder();
+      const pass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: colorAttachmentView,
+            clearValue: { r: 0.4, g: 1.0, b: 0.0, a: 1.0 },
+            loadOp: 'clear',
+            storeOp: 'store',
+          },
+        ],
+      });
+      pass.end();
+      device.queue.submit([encoder.finish()]);
+    }
+  }
+
+  function drawAll() {
+    draw('cvs00', 'opaque', { reconfigureAfterLost: false, drawAfterLost: false });
+    draw('cvs01', 'opaque', { reconfigureAfterLost: false, drawAfterLost: true });
+    draw('cvs02', 'premultiplied', { reconfigureAfterLost: false, drawAfterLost: false });
+    draw('cvs03', 'premultiplied', { reconfigureAfterLost: false, drawAfterLost: true });
+
+    draw('cvs10', 'opaque', { reconfigureAfterLost: true, drawAfterLost: false });
```

Comment:
I think `cvs10` should display transparent color.

Reply:
Hm, why is that? I think the spec says, reconfiguring should `Replace the drawing buffer` (which sets the drawing buffer to transparent black), then when the canvas is presented, `get a copy of the image contents of a context` clears the alpha to 0.

Reply:
My understanding is that although reconfiguring has set the drawing buffer to black, but drawAfterLoss is false(getCurrentTexture() is not called), canvas is not redrawn and it is no content to presented, so it should be transparent.
For cvs11, because drawAfterLost is true, it should be displayed in black.

Reply:
According to the spec, configure() also replaces the drawing buffer with a blank image (as do context creation, resize, unconfigure(), getCurrentTexture(), and transferToImageBitmap()).

Which makes me realize there should probably be some tests for unconfigure as well. I was going to add some but didn't think of why it would be useful.